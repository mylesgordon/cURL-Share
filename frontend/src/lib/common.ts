export function splitAndTrim(input: string): Array<string> {
		return input.split(',').map((item) => item.trim());
	}