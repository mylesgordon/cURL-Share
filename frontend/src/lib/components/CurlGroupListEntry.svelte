<script lang="ts">
	import { Card, Tag } from 'agnostic-svelte';

	export let description: string;
	export let groupId: number;
	export let labels: Array<string>;
	export let projectId: string;
	export let title: string;

	const dataTestIdPrefix = `curl-group-list-entry-${groupId}`;
	const link = `/project/${projectId}/group/${groupId}`;
</script>

<Card isBorder isRounded isShadow css="mt-4">
	<a
		href={link}
		class="p-4 flex-grow full truncate text-center project-title"
		data-testid={dataTestIdPrefix + '-title'}
	>
		{title}
	</a>
	<p class="p-4 flex-grow project-description" data-testid={dataTestIdPrefix + '-description'}>
		{description}
	</p>
	<div
		aria-label="Project labels"
		class="flex flex-grow flex-wrap justify-center project-label-container"
	>
		{#each labels as label}
			<span data-testid={dataTestIdPrefix + '-label-' + label}>
				<Tag shape="pill">{label}</Tag>
			</span>
		{/each}
	</div>
</Card>

<style>
	/* TODO: figure out tailwind ignoring classes */
	.project-title {
		width: 8rem;
	}
	.project-description {
		width: 32rem;
	}
	.project-label-container {
		width: 14rem;
		gap: 0.5rem;
		padding: 1rem 0 1rem 1rem;
	}
</style>
