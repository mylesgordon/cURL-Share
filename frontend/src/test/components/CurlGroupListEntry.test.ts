import CurlGroupListEntry from '$lib/components/CurlGroupListEntry.svelte';
import { screen } from '@testing-library/dom';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/svelte';
import { testMatchingSnapshot } from '../common';
import type { ComponentProps } from 'svelte';

describe('ProjectListEntry component', () => {
	const props: ComponentProps<CurlGroupListEntry> = {
		groupId: 1,
		projectId: '1',
		title: 'Login Journey',
		description: 'cURL group defining a user journey!',
		labels: ['label 1', 'label 2']
	};

	testMatchingSnapshot(CurlGroupListEntry, props);

	it('should render title, description and labels', () => {
		render(CurlGroupListEntry, props);
		const description = screen.getByTestId('curl-group-list-entry-1-description');
		const title = screen.getByTestId('curl-group-list-entry-1-title');
		const label1 = screen.getByTestId('curl-group-list-entry-1-label-label 1');
		const label2 = screen.getByTestId('curl-group-list-entry-1-label-label 2');
		expect(description).toHaveTextContent(props.description);
		expect(label1).toHaveTextContent(props.labels[0]);
		expect(label2).toHaveTextContent(props.labels[1]);
		expect(title).toHaveTextContent(props.title);
	});

	it('should link to the correct place', () => {
		render(CurlGroupListEntry, props);
		const title = screen.getByTestId('curl-group-list-entry-1-title');
		expect(title.getAttribute('href')).toBe('/project/1/group/1');
	});
});
