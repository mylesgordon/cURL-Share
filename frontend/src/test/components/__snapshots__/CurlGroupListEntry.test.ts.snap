// Vitest Snapshot v1

exports[`ProjectListEntry component > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="card card-shadow card-rounded card-border mt-4 s-CTyDWs24qi3Z"
    >
      <a
        class="p-4 flex-grow full truncate text-center project-title s-kSbj4w364X09"
        data-testid="curl-group-list-entry-1-title"
        href="/project/1/group/1"
      >
        Login Journey
      </a>
       
      <p
        class="p-4 flex-grow project-description s-kSbj4w364X09"
        data-testid="curl-group-list-entry-1-description"
      >
        cURL group defining a user journey!
      </p>
       
      <div
        aria-label="Project labels"
        class="flex flex-grow flex-wrap justify-center project-label-container s-kSbj4w364X09"
      >
        <span
          class="s-kSbj4w364X09"
          data-testid="curl-group-list-entry-1-label-label 1"
        >
          <span
            class="tag tag-pill s-1BpX-UQ-3FT-"
          >
            label 1
          </span>
          <!--&lt;Tag&gt;-->
           
        </span>
        <span
          class="s-kSbj4w364X09"
          data-testid="curl-group-list-entry-1-label-label 2"
        >
          <span
            class="tag tag-pill s-1BpX-UQ-3FT-"
          >
            label 2
          </span>
          <!--&lt;Tag&gt;-->
           
        </span>
      </div>
    </div>
    <!--&lt;Card&gt;-->
    <!--&lt;CurlGroupListEntry&gt;-->
  </div>
</body>
`;
